### PR TITLE
splitting waveform in whisper offline

### PR DESCRIPTION
**Pull request by lusk4r**
- new pad adaptation for n_frames > 3000 in prep_audio
- splitting strategy of the waveform (simplified version of the strategy proposed [here](https://github.com/openai/whisper/blob/b38a1f20f4b23f3f3099af2c3e0ca95627276ddf/whisper/transcribe.py#L226))

### CODE REVIEW

**Comment by geohot** [comment]
I don't understand what this does. Is it tested?

**Comment by lusk4r** [comment]
It is an extension of what was done in #2042 (for the offline version of whisper), solving issue #2048 caused by audio files longer than 30 seconds. 

I tried to implement a similar version to what was done in the [transcribe method of whisper](https://github.com/openai/whisper/blob/b38a1f20f4b23f3f3099af2c3e0ca95627276ddf/whisper/transcribe.py#L226) by managing the pad in case the number of frames of the log_spec in method prep_audio was > 3000 (due to recordings > 30s) and passing the chunks obtained from the starting log_spec to the encoder. 

I used the test written in #2042 for audio < 30s (it works correctly), added tests to verify the pad, and a test for audio > 30s (in which I enlarged the test.wav file).

File: examples/whisper.py [hunk 1]
```diff
@@ -214,30 +212,42 @@ def init_whisper(model_name="tiny.en"):
...
+      out = model.decoder(Tensor([lst]), dat).realize()
+      idx = int(out[0,-1].argmax().numpy().item())
+      lst.append(idx)
+      transcription = enc.decode(lst)
+      print(transcription)
+      if lst[-1] == enc._special_tokens["<|endoftext|>"]:
+        final_transcription += transcription
+        if seek == log_spec.shape[-1]:
+          return  final_transcription
+        lst = [enc._special_tokens["<|startoftranscript|>"], enc._special_tokens["<|notimestamps|>"]]
```
Comment (mmmkkaaayy) [inline examples/whisper.py:239]: Is this done in the reference impl?

Comment (lusk4r) [reply examples/whisper.py]: good point! for semplicity in the previous implementation I followed [this scenario](https://github.com/openai/whisper/blob/b38a1f20f4b23f3f3099af2c3e0ca95627276ddf/whisper/transcribe.py#L366).

In the current version, I have instead implemented the token feed

**Comment by geohot** [comment]
Is this covered by #2074?

**Comment by mmmkkaaayy** [comment]
#2074 doesn't contain this functionality (the OpenAI evaluations truncated all samples to 30s). But it might be better to wait until we've gotten an optimized/JITted impl before trying to include this.

**Comment by lusk4r** [comment]
> #2074 doesn't contain this functionality (the OpenAI evaluations truncated all samples to 30s). But it might be better to wait until we've gotten an optimized/JITted impl before trying to include this.

agree with this. Provisionally, it might make sense to check the size of the input audio and cut it if greater than 30s

**Pull request closed without merging**